src/components/ImageWithAspect.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+import React, { Component } from "react";
+import { Image, View } from "react-native";
+
+type Props = {
+  ratio: number,
+  source: Image.propTypes.source
+};
+
+type State = {
+  height: number
+};
+
+class ImageWithAspect extends Component<Props, State> {
+  constructor() {
+    super();
+    this.state = {
+      height: 0
+    };
+  }
+  handleLayout = (event: { nativeEvent: { layout: { width: number } } }) => {
+    this.setState({
+      height: event.nativeEvent.layout.width / this.props.ratio
```

Comment:
There is a style prop called [`aspectRatio`](http://facebook.github.io/react-native/docs/layout-props.html#aspectratio). Did you try using this?

Reply:
Amazing, I didn't even know this existed. I have updated using this instead.